### DISCUSSION TITLE
unsupported: with expression in select statement

### DISCUSSION THREAD

**Discussion by mrtnvh** [planetscale/discussion#270] [Q&A] [answered]
Hi,

When running a recursive CTE, I get the following error message.

```
unsupported: with expression in select statement
```

Is PlanetScale able to or planning support recursive CTEs?

**Comment by fooforge** [comment, marked as answer]
@vanhoofmaarten I'm afraid, no. [Vitess does not support](https://vitess.io/docs/14.0/reference/compatibility/mysql-compatibility/#window-functions-and-ctes) window functions or common table expressions yet. We're actively working on closing the compatibility gap, but I cannot say if or when we're going to get to CTEs at this time.

**Reply by mrtnvh** [reply]
Oh shucks. Thanks for the answer!

**Comment by Brian-1150** [comment]
@fooforge  I am wondering what the CTE content on the PlanetScale site is all about then, since it is not even supported?
https://planetscale.com/learn/courses/mysql-for-developers/queries/common-table-expressions-ctes?autoplay=1

**Reply by orware** [reply]
Hello @Brian-1150,

The course covers MySQL 8 more generally so it does go over functionality available within MySQL 8, even if it may not necessarily be supported within PlanetScale yet. This was done mainly due to the educational nature of that content.

In the context of a PlanetScale database, where you have MySQL 8 plus Vitess running on top of it, we have the additional factor of not all newer MySQL 8 capabilities / functions being supported just yet since those require extra support to be built-in at the Vitess layer, which occurs more slowly over time.

**Reply by rsimonton** [reply]
This is just wildly misleading. At a bare minimum the course materials should explicitly state that those features are not supported by PlanetScale.  Our team lost a bunch of time trying to get to the bottom of why we couldn't get CTEs to work _given that the PlanetScale course indicates they're available, and even provides sample queries_.
